src/dayview.ts

```diff
@@ -424,19 +443,23 @@ export class DayViewComponent implements ICalendarComponent, OnInit, OnChanges,
                 allDayEvents.push({
                     event
                 });
+                if (event.category && this.dayviewCategorySource && this.dayviewCategorySource.has(event.category)) {
+                    displayCategorySet.add(event.category);
```

Comment:
do you still need displaycategorySet? dayviewcategorySource controls all the events should be displayed?

Reply:
`displaycategorySet` is still required to display categories only for current day. If we use `dayviewcategorySource`, all categories will display, there will be many empty event columns from other days.